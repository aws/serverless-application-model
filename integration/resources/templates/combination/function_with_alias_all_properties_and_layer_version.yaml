Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: ${codeuri}
      Handler: index.handler
      Runtime: nodejs18.x
      AutoPublishAlias: Live
      AutoPublishAliasAllProperties: true
      Layers:
      - !Ref MyLayer

  MyLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      ContentUri: ${contenturi}
      RetentionPolicy: Delete
      PublishLambdaVersion: true
      Description: test
Metadata:
  SamTransformTest: true
