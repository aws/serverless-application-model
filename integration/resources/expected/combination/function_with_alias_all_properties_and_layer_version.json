[
  {
    "LogicalResourceId": "MyLambdaFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionAliasLive",
    "ResourceType": "AWS::Lambda::Alias"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionVersion",
    "ResourceType": "AWS::Lambda::Version"
  },
  {
    "LogicalResourceId": "MyLayer",
    "ResourceType": "AWS::Lambda::LayerVersion"
  }
]
